index.ts

```diff
@@ -57,22 +49,23 @@ let interrupted = false;
  * Calls all interrupt handlers, then exits with exit code 0.
  *
  * If called more than once it skips waiting for the interrupt handlers to
- * finish and exits with exit code 1.
+ * finish and exits with exit code 1. If there are any errors with interrupt
+ * handlers, the process exits immediately with exit code 2.
  *
  * This function is called when a SIGINT is received.
  */
 export function interrupt(): void {
-  if (interruptHandlers.length === 0) {
-    return process.exit(0);
-  }
   if (interrupted) {
-    console.log('\nKilling process with extreme prejudice');
+    console.log('\nGot multiple interrupts, exiting immediately!');
     return process.exit(1);
-  } else {
-    interrupted = true;
   }
+  interrupted = true;
 
-  close(() => process.exit(0));
+  close().then(() => process.exit(0), (error) => {
```

Comment:
It could, but a user could get that effect with code like this:

```javascript
cleankill.onInterrupt(() => { 
  // code that would run after the proposed interrupt() promise resolves
});
cleankill.interrupt();
```

If we return a promise from `interrupt` we'd need to carefully explain that it would only resolve if this was the first call to `interrupt`. Seems like unnecessary complexity in the API.